x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/add_domain/add_domain_logic.test.ts

```diff
@@ -277,32 +366,340 @@ describe('AddDomainLogic', () => {
         mount({ addDomainFormInputValue: 'https://swiftype.com/site-search' });
-        jest.spyOn(AddDomainLogic.actions, 'onValidateDomain');
+        jest.spyOn(AddDomainLogic.actions, 'validateDomainInitialVerification');
 
-        AddDomainLogic.actions.validateDomain();
+        AddDomainLogic.actions.startDomainValidation();
         await nextTick();
 
-        expect(AddDomainLogic.actions.onValidateDomain).toHaveBeenCalledWith(
+        expect(AddDomainLogic.actions.validateDomainInitialVerification).toHaveBeenCalledWith(
           'https://swiftype.com',
           '/site-search'
         );
         expect(getDomainWithProtocol).toHaveBeenCalledWith('https://swiftype.com');
       });
     });
+
+    describe('validateDomainInitialVerification', () => {
+      it('validates the url', async () => {
+        jest.spyOn(AddDomainLogic.actions, 'performDomainValidationStep');
+
+        AddDomainLogic.actions.validateDomainInitialVerification('https://elastic.co', '/');
+        await nextTick();
+
+        expect(
+          AddDomainLogic.actions.performDomainValidationStep
+        ).toHaveBeenCalledWith('initialValidation', ['url']);
+      });
+    });
+
+    describe('validateDomainContentVerification', () => {
+      it('validates the domain content', async () => {
+        jest.spyOn(AddDomainLogic.actions, 'performDomainValidationStep');
+
+        AddDomainLogic.actions.validateDomainContentVerification();
+        await nextTick();
+
+        expect(
+          AddDomainLogic.actions.performDomainValidationStep
+        ).toHaveBeenCalledWith('contentVerification', ['url_request', 'url_content']);
+      });
+    });
+
+    describe('validateDomainIndexingRestrictions', () => {
+      it("validates the domain's robots.txt", async () => {
+        jest.spyOn(AddDomainLogic.actions, 'performDomainValidationStep');
+
+        AddDomainLogic.actions.validateDomainIndexingRestrictions();
+        await nextTick();
+
+        expect(
+          AddDomainLogic.actions.performDomainValidationStep
+        ).toHaveBeenCalledWith('indexingRestrictions', ['robots_txt']);
+      });
+    });
+
+    describe('validateDomainNetworkConnectivity', () => {
+      it("validates the domain's dns", async () => {
+        jest.spyOn(AddDomainLogic.actions, 'performDomainValidationStep');
+
+        AddDomainLogic.actions.validateDomainNetworkConnectivity();
+        await nextTick();
+
+        expect(
+          AddDomainLogic.actions.performDomainValidationStep
+        ).toHaveBeenCalledWith('networkConnectivity', ['dns', 'tcp']);
+      });
+    });
+
+    describe('performDomainValidationStep', () => {
+      beforeEach(() => {
+        mount({
+          addDomainFormInputValue: 'https://elastic.co',
+        });
+      });
+
+      describe('on success', () => {
+        it('sets all remaining validation steps invaid on a blocking failure', async () => {
```

Comment:
[not a change request] I think I get it but this threw me for a loop honestly to have a "blocking failure" inside an "on success" block.

I think another way of organizing this that could make sense would be grouping it by the end-user experience rather than strictly by the technical details/flow. Something like,

```js
describe('happy path with no blocking failures', () => {
  it('updates the validation result')
  describe('validation chain')
});

describe('failures', () => {
  it('sets all remaining validation steps invalid on a blocking failure')
  it('sets all remaining validation steps invalid on API error')
 // one nice thing about having these two tests next to each other is that it's easier to compare/contrast the different but similar responses
});
```

WDYT? Possible this is worse if it doesn't follow the code path/branching as closely, so def could be a developer experience vs user experience type thing. 🤷 feel free to pass on this for now / unless you really like it

Reply:
Technically a blocking failure _is_ a happy path?  As far as this component is concerned it is a valid response from the server.  When I think of "on error" in this situations I'm usually thinking of the logic inside of our `} catch(e) {` blocks when the HTTP request returns with a 50X or similar.

Reply:
No worries - like I said, I was approaching this from a "user" happy path rather than a "developer" happy path. I think both ways of thinking are legit, so if it doesn't make sense to you I'm fine either way